tests/lib/file-finder.js

```diff
@@ -160,7 +160,7 @@ describe("FileFinder", () => {
             it("should only find one package.json from the root", () => {
                 expected = path.join(process.cwd(), "package.json");
                 finder = new FileFinder("package.json", process.cwd());
-                actual = finder.findAllInDirectoryAndParents(fileFinderDir);
+                actual = Array.from(finder.findAllInDirectoryAndParents(fileFinderDir)).slice(-1);
 
                 /**
                  * Filter files outside of current workspace, otherwise test fails,
```

Comment:
Can you add a test for this change to make sure the parent directories are not traversed?

One way to do this would be to create a config file with `root: true` somewhere in `tests/fixtures/config-file`, (e.g. `tests/fixtures/parent-traversal/foo/.eslintrc.json`), and have a broken config file (e.g. with invalid JSON) in a parent directory (e.g. `tests/fixtures/parent-traversal/.eslintrc.json`). Then `getLocalConfig` should be able to resolve the config for `tests/fixtures/parent-traversal/foo/bar.js` rather than throwing an error for an invalid config, because it will never reach the file with invalid JSON.